flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/agg/AggsHandlerCodeGenerator.scala

```diff
@@ -1082,6 +1100,17 @@ class AggsHandlerCodeGenerator(
     }
   }
 
+  private def getShiftEpochMills(itemExpr: String): String = {
+     if ("UTC".equals(shiftTimeZone.getId)) {
+       itemExpr
+     } else {
+       s"""
+          |$TIME_WINDOW_UTIL.toEpochMills(
+          |$itemExpr, java.time.ZoneId.of("${shiftTimeZone.getId}"))
```

Comment:
nit: Personally, I suggest to serialize `ZoneId` as a member field to avoid construct the `ZoneId` per record.